hypothetical Poisson-distributed data related to pet tigers. The Poisson distribution is often used to model the number of events occurring within a fixed interval of time or space, given a constant average rate of occurrence.


Scenario: Suppose we are interested in the number of new pet tigers that are adopted each month by a specific animal shelter. We assume that the average rate at which new pet tigers are adopted is 3 per month. We will use the Poisson distribution to generate data for 12 months.

Parameters
Average rate (λ): 3 pet tigers per month
Data Generation
Using the Poisson distribution with λ = 3, we generate the number of new pet tigers adopted each month for a year. Here is some simulated data:
Month
New Pet Tigers Adopted
1
2
2
4
3
3
4
1
5
5
6
3
7
2
8
6
9
3
10
4
11
2
12
7

This dataset is generated with the Poisson distribution in mind, but the specific values can vary each time you generate new data because Poisson data is inherently random. The mean number of new adoptions per month should approximate to the rate parameter (λ = 3) over many samples.


In [2]:
import pandas as pd
import numpy as np
import statsmodels.api as sm
import statsmodels.formula.api as smf


In [3]:
# Step 1: Prepare the data
data = {
    'Month': np.arange(1, 13),
    'New_Pet_Tigers_Adopted': [2, 4, 3, 1, 5, 3, 2, 6, 3, 4, 2, 7]
}

df = pd.DataFrame(data)

In [4]:

# Step 2: Fit the Poisson GLM
# Define the model formula
model = smf.poisson('New_Pet_Tigers_Adopted ~ Month', data=df)


In [5]:
# Fit the model
results = model.fit()

Optimization terminated successfully.
         Current function value: 1.858374
         Iterations 5


In [6]:
# Step 3: Print the summary of the model
print(results.summary())

                            Poisson Regression Results                            
Dep. Variable:     New_Pet_Tigers_Adopted   No. Observations:                   12
Model:                            Poisson   Df Residuals:                       10
Method:                               MLE   Df Model:                            1
Date:                    Thu, 04 Jul 2024   Pseudo R-squ.:                 0.03405
Time:                            20:31:48   Log-Likelihood:                -22.300
converged:                           True   LL-Null:                       -23.087
Covariance Type:                nonrobust   LLR p-value:                    0.2099
                 coef    std err          z      P>|z|      [0.025      0.975]
------------------------------------------------------------------------------
Intercept      0.8675      0.359      2.417      0.016       0.164       1.571
Month          0.0564      0.045      1.247      0.212      -0.032       0.145


In [7]:
results.summary()

<class 'statsmodels.iolib.summary.Summary'>
"""
                            Poisson Regression Results                            
==================================================================================
Dep. Variable:     New_Pet_Tigers_Adopted   No. Observations:                   12
Model:                            Poisson   Df Residuals:                       10
Method:                               MLE   Df Model:                            1
Date:                    Thu, 04 Jul 2024   Pseudo R-squ.:                 0.03405
Time:                            20:34:55   Log-Likelihood:                -22.300
converged:                           True   LL-Null:                       -23.087
Covariance Type:                nonrobust   LLR p-value:                    0.2099
==============================================================================
                 coef    std err          z      P>|z|      [0.025      0.975]
------------------------------------------------------------------------------
Intercept      0.8675      0.359      2.417      0.016       0.164       1.571
Month          0.0564      0.045      1.247      0.212      -0.032       0.145
==============================================================================
"""

In [8]:
import pandas as pd
import numpy as np
import statsmodels.api as sm
import statsmodels.formula.api as smf

# Step 1: Prepare the data
data = {
    'Month': np.arange(1, 13),
    'New_Pet_Tigers_Adopted': [2, 4, 3, 1, 5, 3, 2, 6, 3, 4, 2, 7]
}

df = pd.DataFrame(data)

# Step 2: Fit the Poisson GLM
# Define the model formula
model = smf.poisson('New_Pet_Tigers_Adopted ~ Month', data=df)

# Fit the model
results = model.fit()

# Step 3: Print the summary of the model
print(results.summary())

Optimization terminated successfully.
         Current function value: 1.858374
         Iterations 5
                            Poisson Regression Results                            
Dep. Variable:     New_Pet_Tigers_Adopted   No. Observations:                   12
Model:                            Poisson   Df Residuals:                       10
Method:                               MLE   Df Model:                            1
Date:                    Thu, 04 Jul 2024   Pseudo R-squ.:                 0.03405
Time:                            20:36:44   Log-Likelihood:                -22.300
converged:                           True   LL-Null:                       -23.087
Covariance Type:                nonrobust   LLR p-value:                    0.2099
                 coef    std err          z      P>|z|      [0.025      0.975]
------------------------------------------------------------------------------
Intercept      0.8675      0.359      2.417      0.016       0.164       1.5

Model Information:
Dep. Variable: The dependent variable (response variable) is New_Pet_Tigers_Adopted, which represents the number of new pet tigers adopted each month.
No. Observations: The number of data points used in the model, which is 12 (one for each month).
Df Residuals: Degrees of freedom for residuals, which is the number of observations minus the number of parameters estimated (here, 12 - 2 = 10).
Df Model: Degrees of freedom for the model, which is the number of predictors (here, 1 for Month).
Model Family: Indicates the model is a Poisson regression.
Method: The method used for fitting the model.
Log-Likelihood: The log-likelihood of the fitted model is -22.300. This value helps assess the model fit but is usually used in comparison to other models.
Pseudo R-squ.: This is a measure of model fit, analogous to R-squared in linear regression. A value of 0.03405 suggests that about 3% of the variability in the response variable is explained by the model.
LL-Null: The log-likelihood of the null model (a model with no predictors), which is -23.087. This is used to compare with the log-likelihood of the fitted model to assess model improvement.
Covariance Type: Indicates the type of covariance used in the standard errors calculations, which is nonrobust in this case.
LLR p-value: The p-value for the likelihood ratio test comparing the fitted model to the null model. A value of 0.2099 indicates that the model does not significantly improve fit over the null model.